edx/edx-enterprise
# [ENT-497] Update user-facing consent view to reflect new data model

**Description:** This PR consolidates the logic in the user-facing data sharing consent view so that there aren't necessarily special paths taken for when we have a course vs. a program for which to grant consent.

**JIRA:** [ENT-497](https://openedx.atlassian.net/browse/ENT-497)

**Dependencies:** https://github.com/edx/edx-platform/pull/16136 & https://github.com/edx/ecommerce/pull/1494

**Merge deadline:** End of week

**Testing instructions:**

1. Check our standard DSC workflows and make sure nothing has changed.
1. To double-check, follow the test instructions on the dependency PRs.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
